bot/bot.py

```diff
@@ -104,6 +107,26 @@ async def wait_until_guild_available(self) -> None:
         """
         await self._guild_available.wait()
 
+    async def login(self, *args, **kwargs) -> None:
+        """Re-create the connector and set up sessions before logging into Discord."""
+        # Use asyncio for DNS resolution instead of threads so threads aren't spammed.
+        self._resolver = aiohttp.AsyncResolver()
+
+        # Use AF_INET as its socket family to prevent HTTPS related problems both locally
+        # and in production.
+        self._connector = aiohttp.TCPConnector(
+            resolver=self._resolver,
+            family=socket.AF_INET,
+        )
+
+        # Client.login() will call HTTPClient.static_login() which will create a session using
+        # this connector attribute.
+        self.http.connector = self._connector
+
+        self.http_session = aiohttp.ClientSession(connector=self._connector)
+
+        await super().login(*args, **kwargs)
+
```

Comment:
I think it would be better, if we moved this logic to the main coroutine, like this:
https://gist.github.com/Rapptz/6706e1c8f23ac27c98cee4dd985c8120#interaction-with-aiohttp

Reply:
Yeah, there are a lot of things that can be redone in a better way by leveraging the new features. I think in the long-term, it'd be better to come up with a base class in bot-core. Thus, I'm indifferent on whether this copied code is updated or left alone until bot-core changes come around.